File: app/src/main/java/getyourcasts/jd/com/getyourcasts/repository/remote/DataRepository.java
package getyourcasts.jd.com.getyourcasts.repository.remote;

import android.content.ContentValues;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.util.List;

import getyourcasts.jd.com.getyourcasts.repository.remote.data.Channel;
import getyourcasts.jd.com.getyourcasts.repository.remote.data.Episode;
import getyourcasts.jd.com.getyourcasts.repository.remote.data.Podcast;


/**
 * Created by chuondao on 8/8/17.
 */


public interface DataRepository {
    @NotNull
    List<Podcast> searchPodcast(@NotNull String var1);

    @Nullable
    Podcast getPodcast(@NotNull String var1);

    @NotNull
    List getAllPodcast();

    @Nullable
    Channel downloadFeed(@NotNull String var1);

    @NotNull
    List<Episode> getAllEpisodesOfPodcast(@NotNull String var1);

    @NotNull
    Episode getEpisode(@NotNull String var1, @NotNull String var2);

    boolean updatePodcast(@NotNull ContentValues var1, @NotNull String var2);

    boolean updateEpisode(@NotNull ContentValues var1, @NotNull Episode var2);

    boolean insertPodcast(@NotNull Podcast var1);

    boolean insertEpisode(@NotNull Episode var1);

    boolean insertEpisodes(@NotNull List<Episode> var1);

    Channel fetchEpisodesFromFeedUrl(String feedUrl);
}

File: app/src/main/java/getyourcasts/jd/com/getyourcasts/repository/DataRepository.kt
package getyourcasts.jd.com.getyourcasts.repository

import android.content.ContentValues
import getyourcasts.jd.com.getyourcasts.repository.remote.data.Channel
import getyourcasts.jd.com.getyourcasts.repository.remote.data.Episode
import getyourcasts.jd.com.getyourcasts.repository.remote.data.FeedItem
import getyourcasts.jd.com.getyourcasts.repository.remote.data.Podcast

/**
 * Repository Interface to encapsulate functionalities of
 * both remote and local repository
 * Created by chuondao on 7/21/17.
 */

//interface DataRepository {
//
//    // query function
//    fun searchPodcast(title: String): List<Podcast>
//
//    fun getPodcast(podcastId: String) : Podcast?
//
//    fun getAllPodcast() : List<Podcast>
//
//    fun downloadFeed(feedUrl: String): Channel?
//
//    fun getAllEpisodesOfPodcast(podcastId: String) : List<Episode>
//
//    fun getEpisode (episodeId: String, podcastId: String): Episode
//
//    // updates function
//    /**
//     * return -1 if update is failed
//     */
//    fun updatePodcast(cv: ContentValues, podcastID: String) : Boolean
//
//    /**
//     * return -1 if update is failed
//     */
//    fun updateEpisode(cv: ContentValues, episode: Episode) : Boolean
//
//    fun insertPodcast(pod: Podcast): Boolean
//
//    fun insertEpisode(episode: Episode): Boolean
//
//    fun insertEpisodes(episodes: List<Episode>) : Boolean
//}
